drivers/UARTSerial.cpp

```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#if DEVICE_SERIAL
+#if (DEVICE_SERIAL && DEVICE_INTERRUPTIN)
```

Comment:
@bridadan Totally agree with you. Probably I should have added this bit in commit message. UARTSerial do use InterruptIn class, but yes in  agreement with you, nobody can use UARTSerial without having interrupts. But just for consistency of the system, added this flag.